know/rdf/parser/wikidata_parser.py

```diff
@@ -1,26 +1,44 @@
 # pylint: disable=no-self-use,unused-import,
 import re
+from qwikidata.sparql import return_sparql_query_results
 from rdf.parser.abstract_parser import AbstractParser, EntityType
+from rdf.parser.sparql_reader import read_sparql
 class WikidataParser(AbstractParser):
     """parses wikidata sources"""
 
     def __init__(self, uri):
         super().__init__(uri)
-        self.entity_id = re.search('Q[0-9]+', self.uri)
+        self.entity_id = re.search('Q[0-9]+', self.uri).group(0)
         if self.entity_id is None:
             raise Exception("Invalid Wikidata URI")
 
     def parse(self) -> dict:
         """call this method to start parsing"""
-        return dict()
+        entity_type =  self.get_entity_type()
+        if entity_type == EntityType.BOOK:
+            return self.parse_book()
+        if entity_type == EntityType.PERSON:
+            return self.parse_person()
+        raise Exception("Unsupported entity type!")
```

Comment:
```suggestion
        entity_type =  self.get_entity_type()
        if entity_type == EntityType.BOOK:
            return self.parse_book()

        if entity_type == EntityType.PERSON:
            return self.parse_person()

        raise Exception("Unsupported entity type!")
```
your lack of empty spacer lines is unsettling